contracts/BaseStrategy.sol

```diff
@@ -734,118 +421,51 @@ abstract contract BaseStrategy {
             }
             debtPayment = debtOutstanding.sub(loss);
         } else {
-            // Free up returns for Vault to pull
             (profit, loss, debtPayment) = prepareReturn(debtOutstanding);
         }
 
-        // Allow Vault to take up to the "harvested" balance of this contract,
-        // which is the amount it has earned since the last time it reported to
-        // the Vault.
+        uint256 pricePerShare = vault.pricePerShare();
```

Comment:
I think it would be better to use profit and loss instead of pricePerShare because pricePerShare won't change immediately due to sandwich protection.

totalDebt = vaul.totalDebt()
require(profit * 1e18 / totalDebt <= totalDebt  * (MAX_BPS - changeLimitRatio)) / MAX_BPS))
require(loss * 1e18 / totalDebt <= totalDebt  * (MAX_BPS - changeLimitRatio)) / MAX_BPS))

something like this. just make the threshold very small. like if we see a 0.1% increase or decrease there is issue

Reply:
if @Grandthrax is a better measure of a big PPS change, by checking individual values (loss and profit) i think we go back to needing individual LimitRatios for each, since profitLimitRatio may be =0.05% but lossLimitRatio may be 0% for most if not all strats

Reply:
Agree with this, except just use each strategies' total debt instead, to keep it more normalized

```solidity
uint256 totalDebt = vaul.strategies(address(self)).totalDebt;
require(profit <= totalDebt * changeLimitRatio / MAX_BPS)
require(loss <= totalDebt * changeLimitRatio / MAX_BPS)
```